ManageIQ/manageiq
# Corrections for converted toolbar buttons

Purpose or Intent
-----------------
1. Group toolbar buttons with same code pattern under one Generic group
2. Fix the issue with vm_retire button [mentioned](https://github.com/ManageIQ/manageiq/pull/10218#issuecomment-239168021) by @dclarizio 
3. Add comment to not forget to remove catch with removing AvailabilityMixin

Links
-----
Parent issue: #6259
Related issue: #6554

Steps for Testing/QA
--------------------
Select any VM, buttons *Set Retirement Date* and *Retire this VM* should work properly. The other changes should not change the behavior
